paper.tex

```diff
@@ -474,11 +466,55 @@ \section{Performance}
+
+In order to stress the GPU with a higher workload, we made two additional test scenes. The first is a very large stroked path with 503,304 individually styled dashed segments which we adapted from Skia's open-source test corpus. We used two variants of this scene with \emph{butt} and \emph{round} cap styles. For the second scene we adapted the \emph{Canvas Paths} test case from MotionMark\cite{todo:mmark}, which renders a large number of randomly generated stroked line, quadratic, and cubic Bézier paths. We made two versions with different sizes: mmark-35k and mmark-60k with 30,000 and 60,000 paths respectively. Table \ref{table:bump-counts} shows the payload sizes of our largest test scenes.
+
+\begin{table}
+    \caption{Comparison of input and output segment counts in the more intensive GPU test cases. \emph{Segments} are input path segments. \emph{Lines} and \emph{Arcs} are the output primitives.}
+    \begin{tabular}{|l|c|c|c|}
+    \hline
+        \multicolumn{1}{|c|}{\textbf{Test}} & \textbf{Input Segments} & \textbf{Lines} & \textbf{Arcs} \\
+    \hline
+        waves.svg & 13,308 & 406,059 & 111,965 \\
+        mmark-35k & 105,022 & 1,039,899 & 500,351 \\
+        mmark-60k & 179,929 & 1,784,896 & 858,162 \\
+        long dash (butt) & 503,304 & 1,060,476 & 1,060,476 \\
+        long dash (round) & 503,304 & 2,013,216 & 1,060,476 \\
+    \hline
+    \end{tabular}
+    \label{table:bump-counts}
+\end{table}
+
+The test scenes were rendered to a GPU texture with dimensions 2088x1600. The contents of each scene were uniformly scaled up (by encoding a transform) to fit the size of the texture, in order to increase the required number of output primitives. As with the CPU timings, we used an error tolerance of 0.25 in all cases. We recorded the execution times across 3,000 GPU submissions for each scene.
+
+\begin{figure}
+    \includegraphics[scale=0.65]{nehab_gpu_timings}
+    \caption{Mean GPU execution times of the compute shader that generates stroked and filled outlines, for the Nehab2020 \emph{timings} data set, in microseconds. The hatched bars show the timings for the version that outputs line primitives while the solid bars show that for arcs. Mali-G78 (the lowest end GPU we tested) converts \emph{waves} (the largest scene in the set) to lines in ~3ms. The compute shader executes at least an order of magnitude faster on the other GPUs.}
+    \Description{A stacked bar chart of the GPU timings for stroke expansion from Nehab2020 timings SVGs}
+    \label{fig:nehab-gpu-timings}
+\end{figure}
+
+\begin{figure}
+    \includegraphics[scale=0.66]{test_scenes_gpu_timings}
+    \caption{Mean GPU execution times of the compute shader for the stress test scenes.}
```

Comment:
The caption should explicitly state the task - which I believe is stroke expansion and flattening here. The description does, but my understanding is that it is not generally shown in the printed copy and is for accessibility.